opentelemetry-api/src/opentelemetry/context/context.py

```diff
@@ -16,6 +16,23 @@
 from abc import ABC, abstractmethod
 
 
+class Context:
+    def __init__(
+        self, values: typing.Optional[typing.Dict[str, object]] = None
+    ):
+        if values is None:
+            values = {}
+        self._data = values
+
+    def get_value(self, key: str) -> "object":
+        return self._data.get(key)
+
+    def snapshot(self) -> typing.Dict[str, object]:
+        if self._data:
+            return dict((key, value) for key, value in self._data.items())
+        return {}
+
+
 class RuntimeContext(ABC):
```

Comment:
I think it may be safer if it is enforced that the instances of this class are singletons. I think we don't want more than one `XRuntimeContext` to exist at the same time. This is just a suggestion until I think of a way to conveniently enforce this :stuck_out_tongue_closed_eyes:

Reply:
Here it is:

```python
from abc import ABC, abstractmethod


class RuntimeContext(ABC):

    _instance = None

    def __new__(cls):
        if RuntimeContext._instance is None:
            RuntimeContext._instance = object.__new__(cls)

        if cls != RuntimeContext._instance.__class__:
            raise Exception(
                "A {} has been instantiated already".format(
                    RuntimeContext._instance.__class__.__name__
                )
            )
        return RuntimeContext._instance

    @abstractmethod
    def set_current(self, context):
        pass


class ThreadLocalRuntimeContext(RuntimeContext):

    def set_current(self, context):
        pass


class ContextVarsContext(RuntimeContext):

    def set_current(self, context):
        pass


print(ThreadLocalRuntimeContext())
print(ThreadLocalRuntimeContext())
print(ThreadLocalRuntimeContext())
print(ContextVarsContext())
```
```
<__main__.ThreadLocalRuntimeContext object at 0x7f5237c30590>
<__main__.ThreadLocalRuntimeContext object at 0x7f5237c30590>
<__main__.ThreadLocalRuntimeContext object at 0x7f5237c30590>
Traceback (most recent call last):
  File "tato.py", line 40, in <module>
    print(ContextVarsContext())
  File "tato.py", line 15, in __new__
    RuntimeContext._instance.__class__.__name__
Exception: A ThreadLocalRuntimeContext has been instantiated already
```
I don't think this should block this PR, it can be opened as an enhancement issue later. Just leaving it here for tracking purposes, feel free to resolve this conversation, @codeboten :slightly_smiling_face: